src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts

```diff
@@ -90,22 +90,24 @@ export var resourceActionsDirective = (
             scope.modals = new Modals($timeout);
             adhPermissions.bindScope(scope, path, "options");
 
-            var badgeAssignmentPoolPath;
-            scope.$watch("resourcePath", (resourcePath) => {
-                if (resourcePath) {
-                    adhHttp.get(resourcePath).then((badgeable) => {
-                        badgeAssignmentPoolPath = badgeable.data[SIBadgeable.nick].post_pool;
-                    });
-                }
-            });
-            adhPermissions.bindScope(scope, () => badgeAssignmentPoolPath, "badgeAssignmentPoolOptions");
-            var params = {
-                depth: 4,
-                content_type: SIBadge.nick
-            };
-            adhHttp.get(scope.resourceWithBadgesUrl, params).then((response) => {
-                scope.badgesExist = response.data[SIPool.nick].count > 0;
-            });
+            if (scope.assignBadges) {
+                var badgeAssignmentPoolPath;
+                scope.$watch("resourcePath", (resourcePath) => {
+                    if (resourcePath) {
+                        adhHttp.get(resourcePath).then((badgeable) => {
+                            badgeAssignmentPoolPath = badgeable.data[SIBadgeable.nick].post_pool;
+                        });
+                    }
+                });
+                adhPermissions.bindScope(scope, () => badgeAssignmentPoolPath, "badgeAssignmentPoolOptions");
+                var params = {
+                    depth: 4,
+                    content_type: SIBadge.nick
+                };
+                adhHttp.get(scope.resourceWithBadgesUrl, params).then((response) => {
+                    scope.badgesExist = response.data[SIPool.nick].count > 0;
+                });
+            }
```

Comment:
This does not work for the case that `scope.assignBadges` switches from false to true after linking. I really thing we should move this logic into the action itself.